utils/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -83,7 +83,9 @@ object OkHttpClientHelper {
      * Apply HTTP proxy settings from the "https_proxy" or "http_proxy" environment variables.
      */
     val applyProxySettingsFromEnv: OkHttpClient.Builder.() -> Unit = {
-        val proxyUrl = Os.env["https_proxy"] ?: Os.env["http_proxy"]
+        val proxyUrl = Os.env["https_proxy"]?.let { if (!it.startsWith("http")) "https://$it" else it }
+            ?: Os.env["http_proxy"]?.let { if (!it.startsWith("http")) "http://$it" else it }
+
         if (proxyUrl != null) {
```

Comment:
Is it necessary to consider the case where `https_proxy` has `http` as protocol?
(I think it isn't...)

Reply:
I'm not sure what you mean... no protocol is added if `https_proxy` has either the `http` or `https` protocol, as both of the latter start with "http". That's IMO how it should be.

Reply:
I meant the case where the wrong protocol is set: 
- http for the 'https_proxy' var
- https for the http_proxy var

Reply:
I had asked myself the same, but I couldn't find any clear specification for this. Interestingly, e.g. https://docs.amazonaws.cn/en_us/cli/latest/userguide/cli-configure-proxy.html has examples that use `http://` for `https_proxy`, so I guess both protocols are allowed for both variables although it does not really make sense to me as of now: How can a HTTP-only proxy be a proxy for a HTTPS connection?

Reply:
Thanks, yeah makes sense. Just wanted to make sure it was considered.